### PR TITLE
fix: unsigned submittable

### PR DESCRIPTION
**Pull request by abdulmth**
Fixes https://github.com/KILTprotocol/ticket/issues/3534

### CODE REVIEW

File: tests/integration/utils.ts [hunk 1]
```diff
@@ -110,7 +110,7 @@ export const devBob = Crypto.makeKeypairFromUri('//Bob')
 export const devCharlie = Crypto.makeKeypairFromUri('//Charlie')
 
 export function addressFromRandom(): KiltAddress {
-  return encodeAddress(randomAsU8a())
+  return encodeAddress(randomAsU8a()) as KiltAddress
```
Comment (rflechtner) [inline tests/integration/utils.ts:113]:
```suggestion
  return encodeAddress(randomAsU8a(), 38)
```
That was a bug that the fix to the `encodeAddress` typings has revealed

File: packages/sdk-js/src/DidHelpers/common.ts [hunk 2]
```diff
@@ -63,3 +73,52 @@ export function convertPublicKey(pk: AcceptedPublicKeyEncodings): {
...
+    | KeyringPair
+    | Blockchain.TransactionSigner
+    | MultibaseKeyPair
+    | KiltAddress
+): {
+  submitterSigner: TransactionSigner | KeyringPair | undefined
+  submitterAccount: KiltAddress
+} {
+  let submitterSigner: TransactionSigner | KeyringPair | undefined
+  let submitterAccount: KiltAddress
```
[the remark this answers is not in the document]
Comment (abdulmth) [reply packages/sdk-js/src/DidHelpers/common.ts]: [e882428](https://github.com/KILTprotocol/sdk-js/pull/890/commits/e88242866285926cec15af2425baf352b6e9ba39)

File: tests/integration/utils.ts (hunk 1 above)
Comment (abdulmth) [reply tests/integration/utils.ts]: [08046d2](https://github.com/KILTprotocol/sdk-js/pull/890/commits/08046d292c3a7c2358c0bdd1a0a000969e76ea22)

File: packages/sdk-js/src/DidHelpers/transact.spec.ts [hunk 3]
```diff
@@ -115,4 +115,49 @@ describe('transact', () => {
...
+      expectedEvents: [
+        { section: 'attestation', method: 'AttestationCreated' },
+      ],
+    }).getSubmittable({ signSubmittable: true })
+
+    expect(txHex).toContain('0x')
+    const parsed = mockedApi.tx(txHex)
+    expect(parsed.method).toHaveProperty('section', 'did')
+    expect(parsed.method).toHaveProperty('method', 'submitDidCall')
+    // todo check signature?
```
[the remark this answers is not in the document]
Comment (abdulmth) [reply packages/sdk-js/src/DidHelpers/transact.spec.ts]: [3d8f14a](https://github.com/KILTprotocol/sdk-js/pull/890/commits/3d8f14a87cb003706e947748b4e6f43c6dc50037)

File: tests/integration/utils.ts (hunk 1 above)
Comment (abdulmth) [reply tests/integration/utils.ts]: [08046d2](https://github.com/KILTprotocol/sdk-js/pull/890/commits/08046d292c3a7c2358c0bdd1a0a000969e76ea22)

File: packages/sdk-js/src/DidHelpers/common.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (rflechtner) [reply packages/sdk-js/src/DidHelpers/common.ts]: very nice

**Review by rflechtner — APPROVED** [review]

**Pull request merged**